lib/index.js

```diff
@@ -241,12 +245,60 @@ Socket.prototype._read = function (n) {}
  * Socket API
  */
 
-Socket.prototype.bind = function (addr) {
-  return this.bound[addr] = this._protect(nn.Bind( this.binding, addr ));
+Socket.prototype.bind = function (addr, fn) {
+  if (fn) return asyncBindWrapper.call(this, nn.asyncBind);
+  var ret;
+  this.sync('Bind', this, addr, ret);
```

Comment:
I don't think we need `ret` in `bind` or `connect`.  We're not using it after invoking `this.sync`.  As far as I can tell, it's `undefined`.  I'm quite sure you can only use Arrays or Objects as in out parameters (reference types).